### PR TITLE
DSG-1871 :  Add date fields , and description to transaction definiti…

### PR DESCRIPTION
**Pull request by deibys-nuvalence**
## Description
JIRA: https://dol-ewf.atlassian.net/browse/DSG-1871

## How Has This Been Tested?
See comment https://dol-ewf.atlassian.net/browse/DSG-1871?focusedCommentId=28455

## Types of changes

Patch
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change (documentation only, will not require redeployment)
- [ ] Refactor or cleanup (functionality unchanged)
Minor
- [ ] New feature (non-breaking change which adds functionality)
Major
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have performed a self-review of my own code
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

- [X] I have confirmed all acceptance criteria defined by the story have been met

### CODE REVIEW

**Review by JayStGelais — APPROVED** [review]

File: service/src/main/java/io/nuvalence/workmanager/service/domain/transaction/TransactionDefinition.java [hunk 1]
```diff
@@ -58,6 +58,9 @@ public class TransactionDefinition extends VersionedEntity implements UpdateTrac
     @Column(name = "name", length = 1024, nullable = false)
     private String name;
 
+    @Column(name = "description")
```
Comment (asultan001) [inline service/src/main/java/io/nuvalence/workmanager/service/domain/transaction/TransactionDefinition.java:61]: can you add the length and nullable properties here also?

File: service/src/main/java/io/nuvalence/workmanager/service/models/TransactionDefinitionFilters.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import java.util.Locale;
+
+import javax.persistence.criteria.Predicate;
+
+/**
+ * Filters for Transaction Definitions.
+ */
+public class TransactionDefinitionFilters {
+
+    private static final String DEFAULT_SORT_BY = "createdTimestamp";
```
Comment (asultan001) [inline service/src/main/java/io/nuvalence/workmanager/service/models/TransactionDefinitionFilters.java:20]: can we sort by name by default instead?

File: swagger.yaml [hunk 3]
```diff
@@ -274,15 +274,17 @@ paths:
...
           schema:
             type: string
+        - $ref: "#/components/parameters/sortByTransactionDefinition"
+        - $ref: "#/components/parameters/sortOrder"
+        - $ref: "#/components/parameters/pageNumber"
+        - $ref: "#/components/parameters/pageSize"
       responses:
         '200':
-          description: Collection of transaction definitions matching search criteria.
+          description: Collection of transaction definition (paged).
```
Comment (asultan001) [inline swagger.yaml:283]: typo here with transaction definitions

File: swagger.yaml [hunk 4]
```diff
@@ -2012,7 +2035,11 @@ components:
     SortByTransaction:
       type: string
       default: createdTimestamp
-      pattern: ^(id|priority|district|createdTimestamp|lastUpdatedTimestamp|status)$
+      pattern: ^(id|priority|district|createdTimestamp|lastUpdatedTimestamp|status|description)$
+    SortByTransactionDefinition:
+      type: string
+      default: createdTimestamp
```
Comment (asultan001) [inline swagger.yaml:2041]: I think name or category may be more reasonable here for a default sort by

**Pull request merged**